challenges/ai-ml/challenge-1.md

```diff
@@ -0,0 +1,250 @@
+# Summary
+
+Implement a CLI tool that detects reconnaisance activity in a body of CloudTrail
+logs.
+
+# Rationale
+
+This exercise has two goals:
+
+* It helps us to understand what to expect from you as a developer, how you
+  write production code, how you reason about API design and how you communicate
+  when trying to understand a problem before you solve it.
+* It helps you get a feel for what it would be like to work at Teleport, as this
+  exercise aims to simulate our day-as-usual and expose you to the type of work
+  we're doing here.
+
+We believe this technique is not only better, but also is more fun compared to
+whiteboard/quiz interviews so common in the industry. It's not without the
+downsides - it could take longer than traditional interviews.
+
+[Some of the best teams use coding challenges.](https://sockpuppet.org/blog/2015/03/06/the-hiring-post/)
+
+We appreciate your time and are looking forward to hack on this project together.
+
+# Levels
+
+There are 6 engineering levels at Teleport. This challenge only supports L4.
+
+Check [Systems Engineering Levels](../../levels/systems.pdf) for more details.
+
+# Interview Process
+
+The interview process will start with you receiving an invite to a private
+Slack channel. That channel will contain the interview panel. You can ask them
+about the engineering culture, work-life balance, or anything else that you
+would like to learn about Teleport.
+
+## Design Doc
+
+Before writing any actual code, we ask that you write a brief design document.
+The design document should cover: design approach, scope, proposed API, security
+considerations, CLI UX, and implementation details where appropriate. Start with
+a brief doc that covers the edge cases and design approach.
+
+Please submit the design document and all code in a GitHub repository. Public
+or private is your choice. Please submit the design document written in
+markdown as a Pull Request to allow us to provide you feedback on the proposed
+design.
+
+A few notes about the design document:
+
+* Try to get the design document approved within the first 2-3 days. This is to
```

Comment:
I’m not sure this aligns well with the goals of the data analyst challenge. Candidates should first spend time exploring the dataset independently-understanding its structure, identifying patterns, and discovering what’s meaningful. Requiring the RFD to be approved within the first 2–3 days feels counterintuitive to a typical data analysis process.